docs/user-guide/self_rewarding.rst

```diff
@@ -0,0 +1,214 @@
+            && echo "---------------" \
+            && echo "Starting training" \
+            && cd ${GPFS} \
+            && export PYTHONPATH="${GPFS}:${PYTHONPATH}" \
+            && export NVTE_ALLOW_NONDETERMINISTIC_ALGO=0 \
+            && export NVTE_MASKED_SOFTMAX_FUSION=0 \
+            && export HYDRA_FULL_ERROR=1 \
+            && python -u ${GPFS}/examples/nlp/gpt/train_gpt_self_rewarding.py \
+               trainer.num_nodes=${SLURM_JOB_NUM_NODES} \
+               trainer.devices=8 \
+               pretrained_checkpoint.restore_from_path='${PRETRAINED_CHECKPOINT_NEMO_FILE}' \
+               "model.data.train_ds.file_path=${TRAIN_DATA_PATH}" \
+               "model.data.validation_ds.file_path=${VALID_DATA_PATH}" \
+               model.micro_batch_size=1 \
+               model.global_batch_size=64 \
+               exp_manager.explicit_log_dir=${RESULTS_DIR} \
+               exp_manager.create_wandb_logger=True \
+               exp_manager.wandb_logger_kwargs.name=${NAME} \
+               exp_manager.wandb_logger_kwargs.project=${PROJECT} \
+               trainer.self_rewarding.max_iterations=3 \
+               trainer.self_rewarding.max_epochs=1 \
+               model.spin.ref_policy_kl_penalty=0.1 \
+               model.spin.use_meta_judge=false \
+               model.spin.length_params.max_length=2048 \
+               model.data.train_ds.max_seq_length=4096
+            EOF
+
+            srun -o $OUTFILE -e $ERRFILE --container-image=$CONTAINER $MOUNTS bash -c "${cmd}"
+            set +x
+
+During Self-Rewarding training, there will be several metrics recorded to WandB which you can monitor, the following of which are specific to Self-Rewarding:
+
+- chosen_lengths: average token length of chosen responses (average taken across GBS)
+- reject_lengths: as above but for rejected responses
+- chosen_generated_rewards: the average reward (across GBS) generated by the LLM-as-a-judge for chosen responses
+- rejected_generated_rewards: as above but for rejected responses
+- rewards_chosen_mean: see below for a definition of what reward means in this context
+- rewards_rejected_mean: as above but for rejected responses
+- bad_samples_per_GBS: the percentage of samples in a GBS which are excluded from training because of bad output from the LLM-as-a-judge (could be caused by parse errors, or all responses being judge with the same score, etc)
+- bad_ends_per_GBS: only valid if using TRT, this tracks the percentage of each GBS where TRT generates incorrect stop tokens (should be really low, < 1%)
+- preference_loss: the raw DPO variant loss
+- sft_loss: if adding an SFT loss (categorical cross-entropy loss) for the chosen response, then you can see that raw loss here
+
+The ``reward`` in this case is calculated as the difference between model log probs and the reference log probs, multiplied by the KL penalty (beta in the original paper), for the ground truth and generated responses.
```

Comment:
Fix punctuation.

The ``reward``, in this case, is calculated as the difference between model log probs and the reference log probs, multiplied by the KL penalty (beta in the original paper), for the ground truth and generated responses.